classes/01class/exercises/c01-aws02/nimmi89/README.md

```diff
@@ -0,0 +1,167 @@
+
+>>>Command to create instance profile and add it to the role.The instance profile allows EC2 to pass the IAM role, YourNewRole, to an EC2 instance. 
+aws iam create-instance-profile --instance-profile-name S3AccessRole-Instance-Profile
+>The command returns:
+{
+    "InstanceProfile": {
+        "Path": "/",
+        "InstanceProfileName": "S3AccessRole-Instance-Profile",
+        "InstanceProfileId": "AIPAWMG45LNQ2WM5DWF5E",
+        "Arn": "arn:aws:iam::438549961569:instance-profile/S3AccessRole-Instance-Profile",
+        "CreateDate": "2020-06-17T08:38:29+00:00",
+        "Roles": []
+    }
+}
+aws iam add-role-to-instance-profile --role-name S3AccessRole --instance-profile-name S3AccessRole-Instance-Profile
+
+>>>Command to attach the instance profile, YourNewRole-Instance-Profile, for the newly created IAM role, YourNewRole, to your EC2 instance, YourInstanceId.
+aws ec2 associate-iam-instance-profile --instance-id i-05d7620963b672ca0 --iam-instance-profile Name=S3AccessRole-Instance-Profile
+>The command returns:
+{
+    "IamInstanceProfileAssociation": {
+        "AssociationId": "iip-assoc-09051d5837a6bb5ba",
+        "InstanceId": "i-05d7620963b672ca0",
+        "IamInstanceProfile": {
+            "Arn": "arn:aws:iam::438549961569:instance-profile/S3AccessRole-Instance-Profile",
+            "Id": "AIPAWMG45LNQ2WM5DWF5E"
+        },
+        "State": "associating"
+    }
+}
+
+>>>Command to verify that the IAM role is now attached to the instance
+aws ec2 describe-iam-instance-profile-associations
+>The command returns:
+{
+    "IamInstanceProfileAssociations": [
+        {
+            "AssociationId": "iip-assoc-0ad4a200b37494d0b",
+            "InstanceId": "i-03db9e088f065c42b",
+            "IamInstanceProfile": {
+                "Arn": "arn:aws:iam::438549961569:instance-profile/cloudwatchrole",
+                "Id": "AIPAWMG45LNQWFJASE25M"
+            },
+            "State": "associated"
+        },
+        {
+            "AssociationId": "iip-assoc-09051d5837a6bb5ba",
+            "InstanceId": "i-05d7620963b672ca0",
+            "IamInstanceProfile": {
+                "Arn": "arn:aws:iam::438549961569:instance-profile/S3AccessRole-Instance-Profile",
+                "Id": "AIPAWMG45LNQ2WM5DWF5E"
+            },
+            "State": "associated"
+        }
+    ]
+}
+>>>SSH into the instance and access your bucket
+ssh -i /home/nimmi/C01-AWS01KeyPair.pem ec2-user@52.64.28.243
+Last login: Wed Jun 17 08:29:16 2020 from 14-201-149-89.static.tpgi.com.au
+
+       __|  __|_  )
+       _|  (     /   Amazon Linux 2 AMI
+      ___|\___|___|
+
+https://aws.amazon.com/amazon-linux-2/
+4 package(s) needed for security, out of 8 available
+Run "sudo yum update" to apply all updates.
+[ec2-user@ip-172-31-32-188 ~]$ aws s3 ls
+2020-06-17 08:16:40 c01-aws02-bucket
+
+S3 Access Successful in the EC2 using IAM Roles!!!!!!!!!!!
```

Comment:
well done!!!

Reply:
To upload a file from instance into s3, s3 full access[aws managed policy] will give unnecessary privileges which are not good for security. So, create a customer-managed policy which allows to put object only and attach it to the role.